### PR TITLE
909 number input fixes

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fix max api number regression and add some time constants

Follow up from #909

### CODE REVIEW

File: client/packages/common/src/ui/components/inputs/TextInput/numeric/PositiveNumberInput.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const PositiveNumberInput = React.forwardRef(
       disabled = false,
       value,
       min = 1,
-      max = Number.MAX_SAFE_INTEGER,
+      max = NumUtils.MAX_SAFE_API_INTEGER,
```
Comment (CarlosNZ) [inline client/packages/common/src/ui/components/inputs/TextInput/numeric/PositiveNumberInput.tsx:20]: So what's wrong with just using `Infinity`?

File: client/packages/common/src/utils/numbers/NumUtils.ts [hunk 2]
```diff
@@ -1,6 +1,11 @@
 const constrain = (value: number, min: number, max: number) =>
   Math.min(Math.max(value, min), max);
 
+const SECOND = 1000;
+const MINUTE = 60 * SECOND;
+const HOUR = 60 * MINUTE;
+const DAY = 24 * HOUR;
```
Comment (CarlosNZ) [inline client/packages/common/src/utils/numbers/NumUtils.ts:7]: Maybe just add comment above clarifying that this is in terms of "milliseconds":

```ts
// from milliseconds:
const SECOND = 1000;
const MINUTE = 60 * SECOND;
const HOUR = 60 * MINUTE;
const DAY = 24 * HOUR;
```

Comment (CarlosNZ) [reply client/packages/common/src/utils/numbers/NumUtils.ts:7]: Oh, I see you kind of did that below. No biggie either way.  :)

File: client/packages/common/src/utils/numbers/NumUtils.ts [hunk 3]
```diff
@@ -12,4 +17,19 @@ export const NumUtils = {
 
     return constrain(parsed, min, max);
   },
+  /**
+   * This constant should be used for values that are potentially send to a backend API that expects
+   * an unsigned 32 bit integer and thus would reject Number.MAX_SAFE_INTEGER.
+   * For example, JS number max size is `2^53 - 1` while the Rust u32 size is `2^32 - 1`.
+   */
+  MAX_SAFE_API_INTEGER: 999999999,
```
Comment (CarlosNZ) [inline client/packages/common/src/utils/numbers/NumUtils.ts:25]: Where does this `999999999` come from? Why not `4294967295` (`2^32 - 1`)?

**Review by CarlosNZ — APPROVED** [review]
👍

See couple of small comments/questions.

File: client/packages/common/src/utils/numbers/NumUtils.ts (hunk 3 above)
Comment (clemens-msupply) [reply client/packages/common/src/utils/numbers/NumUtils.ts:25]: This was the previous value... @mark-prins was this an arbitrary choice?

File: client/packages/common/src/utils/numbers/NumUtils.ts (hunk 2 above)
Comment (clemens-msupply) [reply client/packages/common/src/utils/numbers/NumUtils.ts:7]: The reason I but the doc comment into NumUtils is that VSCode doesn't show it when hovering NumUtils.DAY when putting it on these "helpers".

image

File: client/packages/common/src/ui/components/inputs/TextInput/numeric/PositiveNumberInput.tsx (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/common/src/ui/components/inputs/TextInput/numeric/PositiveNumberInput.tsx:20]: there is no special infinity number in an unsigned int :-)

File: client/packages/common/src/utils/numbers/NumUtils.ts (hunk 3 above)
Comment (mark-prins) [reply client/packages/common/src/utils/numbers/NumUtils.ts:20]: yes, iirc just an arbitrary big number. when we created the controls we were dealing with pack sizes and quantities, and reasoned that it is unlikely anyone will order `2^32 - 1` packs 😉

**Pull request merged**